### PR TITLE
[Refactoring] Moving mailbox-store test classes to JUnit 5 (Part 2)

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: mailbox/elasticsearch/pom.xml [hunk 1]
```diff
@@ -89,6 +89,12 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-guice-common</artifactId>
```
Comment (chibenwa) [inline mailbox/elasticsearch/pom.xml]: Why is this needed?

This dependency feels bad...

File: mailbox/tika/pom.xml [hunk 2]
```diff
@@ -40,6 +40,12 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>apache-james-mailbox-store</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-guice-common</artifactId>
```
Comment (chibenwa) [inline mailbox/tika/pom.xml]: Idem, this is a bad thing...

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TestTikaModule.java [hunk 3]
```diff
@@ -17,21 +17,16 @@
...
-import org.apache.james.mailbox.tika.TikaConfiguration;
-import org.apache.james.mailbox.tika.TikaContainer;
-import org.apache.james.mailbox.tika.TikaHttpClient;
-import org.apache.james.mailbox.tika.TikaHttpClientImpl;
-
 import com.google.inject.AbstractModule;
 import com.google.inject.Provides;
 
-public class TestTikaModule extends AbstractModule {
+class TestTikaModule extends AbstractModule {
```
Comment (chibenwa) [inline mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TestTikaModule.java]: Why do we need that here?

File: mailbox/elasticsearch/pom.xml (hunk 1 above)
Comment (Arsnael) [reply mailbox/elasticsearch/pom.xml]: Not sure myself, but the compiler wouldn't stop crying that it's missing this overwise since I switched to the `TikaExtension` that is using `GuiceModuleTestExtension`... But yes there is probably better to do here than just the copy/paste of something existing

**Comment by Arsnael** [comment]
> I think there is a design issue with your proposal.
> 
> Maybe more things can be extracted to the backends?

Yes you might be right... I just wanted to reuse the `TikaExtension` used for integration tests with the james guice cassandra server, but maybe this is overkill... will think of something else

**Comment by chibenwa** [comment]
We need the TikaExtension in mailbox-tika

We need the ES extension in backends-es

**Comment by trantienduchn** [comment]
read it

**Comment by Arsnael** [comment]
> We need the TikaExtension in mailbox-tika
> 
> We need the ES extension in backends-es

And that's where they are already :)

**Comment by Arsnael** [comment]
```
[387f982c270a0c5f25d55ad93d4fbf5ec89f0345] 10:11:16.022 [ERROR] spark.Spark - ignite failed
[387f982c270a0c5f25d55ad93d4fbf5ec89f0345] java.net.BindException: Address already in use
```
test this please

**Comment by chibenwa** [comment]
`FileMailQueueTest.dequeueShouldBeConcurrent » ConditionTimeout Condition with ...`

test this please

**Comment by Arsnael** [comment]
```
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR] Errors: 
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   IndexableMessageTest.setUp:71 » IllegalState Mapped port can only be obtained ...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [ERROR]   MessageToElasticSearchJsonTest.setUp:81 » IllegalState Mapped port can only be...
[d8b7fec43b2e9c6af06d9d5690bb52dd4803415a] [INFO] 
```
seems there is some conflicts between the Tika extension using the singleton rule in `ElasticSearchIntegrationTest` and the singleton rule used as a `ClassRule` in `IndexableMessageTest` and `MessageToElasticSearchJsonTest`.

It seems also that the tika singleton container is only being used by those 3 test classes. I am thinking of maybe extracting the Tika extension work in a new PR where I would migrate the corresponding test classes to JUnit 5 using Tika extension, and writing a new better singleton Tika container class. 

If you have a better idea, don't hesitate :)

**Comment by Arsnael** [comment]
Tika work has been moved to #2976 , thus only one commit is left here

File: mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java [hunk 4]
```diff
@@ -18,22 +18,55 @@
...
+    void setUp() throws Exception {
+        aliceSession = MailboxSessionUtil.create(ALICE);
+        bobSession = MailboxSessionUtil.create(BOB);
+        MessageManagerTestSystem testSystem = createTestSystem();
+        mailboxManager = testSystem.getMailboxManager();
+
+        testSystem.createMailbox(INBOX_ALICE, aliceSession);
+        testSystem.createMailbox(MailboxFixture.OUTBOX_ALICE, aliceSession);
+        testSystem.createMailbox(MailboxFixture.SENT_ALICE, aliceSession);
+        testSystem.createMailbox(MailboxFixture.INBOX_BOB, bobSession);
```
Comment (chibenwa) [inline mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java]: Such a setUp content have nothing to do in the implementation

File: mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java [hunk 5]
```diff
@@ -18,22 +18,55 @@
...
-public class InMemoryMessageManagerTest extends AbstractMessageManagerTest {
+class InMemoryMessageManagerTest implements MessageManagerContract {
 
-    @Override
-    @Before
-    public void setUp() throws Exception {
-        super.setUp();
+    MailboxManager mailboxManager;
+    MailboxSession aliceSession;
+    MailboxSession bobSession;
```
Comment (chibenwa) [inline mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java]: Idem, these fields (sessions) should be in the contract.

File: mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java (hunk 4 above)
Comment (Arsnael) [reply mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java]: I think I had a test isolation issue when I tried to do like this, but will dig more in that today

File: mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java (hunk 5 above)
Comment (Arsnael) [reply mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java]: Well the issue is that it cannot create a new mailbox in the second test if the sessions are being initialized in the contract... Anyway we decided after discussion to keep the abstract class and drop the interface conversion, as it will keep the code simple.

**Comment by chibenwa** [comment]
Was :green_apple:

**Comment by Arsnael** [comment]
Merged
